src/backend/protocol.ts

```diff
@@ -2,91 +2,177 @@ import { dialog } from 'electron'
+      `Could not receive game data for ${arg}!`,
+      LogPrefix.ProtocolHandler
+    )
+  }
+
+  const { is_installed, title, app_name, runner: gameRunner } = game
+
+  if (!is_installed) {
+    logInfo(`"${title}" not installed.`, LogPrefix.ProtocolHandler)
+
+    if (!mainWindow) {
+      return
     }
 
-    const { is_installed, title, app_name, runner } = game
-    if (!is_installed) {
-      logInfo(`"${arg}" not installed.`, LogPrefix.ProtocolHandler)
-
-      if (!mainWindow) {
-        return
-      }
-
-      const { response } = await dialog.showMessageBox(mainWindow, {
-        buttons: [i18next.t('box.yes'), i18next.t('box.no')],
-        cancelId: 1,
-        message: `${title} ${i18next.t(
-          'box.protocol.install.not_installed',
-          'Is Not Installed, do you wish to Install it?'
-        )}`,
-        title: title
+    const { response } = await dialog.showMessageBox(mainWindow, {
+      buttons: [i18next.t('box.yes'), i18next.t('box.no')],
+      cancelId: 1,
+      message: `${title} ${i18next.t(
+        'box.protocol.install.not_installed',
+        'Is Not Installed, do you wish to Install it?'
+      )}`,
+      title: title,
+      icon: icon
+    })
+    if (response === 0) {
+      return sendFrontendMessage('installGame', {
+        appName: app_name,
+        runner: gameRunner
       })
-      if (response === 0) {
-        const { filePaths, canceled } = await dialog.showOpenDialog({
-          buttonLabel: i18next.t('box.choose'),
-          properties: ['openDirectory'],
-          title: i18next.t('install.path', 'Select Install Path')
-        })
-        if (canceled) {
-          return
-        }
-        if (filePaths[0]) {
-          return sendFrontendMessage('installGame', {
-            appName: app_name,
-            runner,
-            path: filePaths[0]
-          })
-        }
-      }
-      if (response === 1) {
-        return logInfo('Not installing game', LogPrefix.ProtocolHandler)
-      }
     }
+    if (response === 1) {
+      return logInfo('Not installing game', LogPrefix.ProtocolHandler)
+    }
+  }
+
+  mainWindow?.hide()
+  sendFrontendMessage('launchGame', arg, gameRunner)
+}
 
-    mainWindow?.hide()
-    sendFrontendMessage('launchGame', arg, runner)
+/**
+ * Finds a game in the runners specified in runnersToSearch
+ * @param runner The runner to search for the game
+ * @param arg The game to search
+ * @returns The game info if found, null otherwise
+ * @example
+ * findGame('gog', '123')
+ * // => { app_name: '123', title: '123', is_installed: true, runner: 'gog' ...}
+ * findGame('legendary', '123')
+ * // => { app_name: '123', title: '123', is_installed: true, runner: 'legendary' ...}
+ **/
+async function findGame(
+  runner: Runner | undefined,
+  arg: string | undefined = ''
+): Promise<GameInfo | SideloadGame | null> {
+  // If the runner is specified, only search for that runner
+  const runnersToSearch = runner ? [runner, 'sideload'] : RUNNERS
```

Comment:
why are we always appending the `sideload` runner? the comment says we'll only check the specified one so it's not clear why we try sideload of the game can't be found in the specified runner (I imagine that check with the sideload runner will always fail anyway in that case)

Reply:
ok, refactor that part now to be more clear.